src/avl.c

```diff
@@ -281,18 +282,21 @@ bool insert_avl_node(avl_pp head, int val)
 
 	if (!root) {
 		root = (avl_p) calloc(1, sizeof(avl_t));
-		root->data = val;
+		root->keyNode = key;
 		*head = root;
 
 		return TRUE;
 	}
-
+	//until current node isn't Null
 	while (root) {
-		if (val < root->data) {
+		//if keyNod < of key to add, go left of tree
+		if (key < root->keyNode) {
+			//if left null, add node as left son
 			if (!root->left) {
 				/* Create an AVL node for new value */
 				root->left = calloc(1, sizeof(avl_t));
-				root->left->data = val;
+				root->left->keyNode = key;
+				root->left->data=data;
```

Comment:
Please put space before and after ` = `.

Reply:
ok, also i discovered littel error in this section, in root check, i wasn't add data on the root node, now correcting it